wpilibc/athena/src/DriverStation.cpp

```diff
@@ -584,9 +582,13 @@ void DriverStation::Run() {
   while (m_isRunning) {
     HAL_WaitForDSData();
     GetData();
+    // notify IsNewControlData variables
+    m_newControlData = true;
```

Comment:
Even if you put it inside the block, the guarantee won't hold true anyway, unless you completely remove the atomic bool, and instead lock around the mutex in isNewControlData as well.